packages/grafana-runtime/src/services/dashboard-srv/types.ts

```diff
@@ -0,0 +1,123 @@
+import { PanelModel as BaseCorePanelModel } from '@grafana/data';
+import { GridPos } from '@grafana/schema';
+
+// Middle types to glue Public Plugin APIs types with internal Grafana types
+export interface CorePanelModel extends BaseCorePanelModel {
+  gridPos: GridPos;
+  refresh: () => void;
+}
+
+export interface CoreDashboardModel {
+  panels: CorePanelModel[];
+  uid: string;
+  title: string;
+  updatePanels: (panels: CorePanelModel[]) => unknown;
+}
+
+export interface CoreDashboardSrv {
+  dashboard?: CoreDashboardModel;
+}
+
+// Public types
+
+/**
+ * Used to interact with the current Grafana dashboard.
+ *
+ * @public
+ */
+export interface PluginsAPIDashboardSrv {
+  /**
+   * @deprecated use `getCurrentDashboard()` instead
+   */
+  dashboard?: PluginsAPIDashboardModel;
+  /**
+   * Represents the currently loaded dashboard. Returns 'undefined' if no dashboard is loaded.
+   */
+  getCurrentDashboard(): PluginsAPIDashboardModel | undefined;
```

Comment:
What do you think about exposing things directly form the `dashboardSrv` instead of returning a model that you then need to interact with?

Reply:
not my preference. I prefer keep the principle of single responsability. dashboardSrv handles dashboardSrv things and returns dashboards, dashboardModel handles dashboard things and return panels. panel model handles panels things and (maybe later) return panels (for the autogenerated panel rows later on)

Reply:
I'm not sure I totally agree. If we are referencing principles you also have the: `Law of Demeter` which we are breaking with the current approach.

Reply:
What is the intention behind leaving all the methods inside `dashboardSrv` instead of leaving them on their respective model?

Reply:
More easy for plugin devs to discover the methods? Less to think about? There's pros and cons to both approaches. 🤷

Reply:
Resolution: We are going for a flat API where all the methods are part of `dashboardSrv`